### PR TITLE
EVEREST-955 | provide an option to clean-up backup storage on cluster/backup deletion

### PR DESCRIPTION
**Pull request by mayankshah1607**

### CODE REVIEW

File: api/database_cluster_backup.go [hunk 1]
```diff
@@ -67,11 +74,34 @@ func (e *EverestServer) CreateDatabaseClusterBackup(ctx echo.Context, namespace
...
+	namespace, name string,
+	params DeleteDatabaseClusterBackupParams,
+) error {
+	cleanupStorage := pointer.Get(params.CleanupBackupStorage)
+	// If cleanup is required, we add a finalizer to the backup object.
+	if cleanupStorage {
+		reqCtx := ctx.Request().Context()
+		backup, err := e.kubeClient.GetDatabaseClusterBackup(reqCtx, namespace, name)
+		if err != nil {
+			return err
```
Comment (michal-kralik) [inline api/database_cluster_backup.go:88]: Shall we return a more user friendly error?

File: api/database_cluster_backup.go [hunk 2]
```diff
@@ -67,11 +74,34 @@ func (e *EverestServer) CreateDatabaseClusterBackup(ctx echo.Context, namespace
...
+		}
+		if err := e.cleanupBackupStorage(reqCtx, backup); err != nil {
+			return err
```
Comment (michal-kralik) [inline api/database_cluster_backup.go:91]: Shall we return a more user friendly error?

File: api/database_cluster_backup.go [hunk 3]
```diff
@@ -67,11 +74,34 @@ func (e *EverestServer) CreateDatabaseClusterBackup(ctx echo.Context, namespace
...
+) error {
+	cleanupStorage := pointer.Get(params.CleanupBackupStorage)
+	// If cleanup is required, we add a finalizer to the backup object.
+	if cleanupStorage {
+		reqCtx := ctx.Request().Context()
+		backup, err := e.kubeClient.GetDatabaseClusterBackup(reqCtx, namespace, name)
+		if err != nil {
+			return err
+		}
+		if err := e.cleanupBackupStorage(reqCtx, backup); err != nil {
```
Comment (michal-kralik) [inline api/database_cluster_backup.go:90]: Shall the backups be removed only after the DB has been successfully removed?

Comment (mayankshah1607) [reply api/database_cluster_backup.go:90]: No, its actually the other way around:

- Add this finalizer on all backups for the DB
- DB deletion is triggered, however there is a finalizer set on it by everest-operator
- DB finalizer will delete all the DBBackups (everest-operator)
- DBBackup finalizer is handled (storage clean-up)
- DB is GC-ed by k8s only after all backups are gone

File: api/database_cluster_backup.go (hunk 1 above)
Comment (mayankshah1607) [reply api/database_cluster_backup.go]: Updated

**Review by michal-kralik — APPROVED** [review]

**Pull request merged**